<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>qSlicerVolumeRenderingModule</class>
 <widget class="qSlicerWidget" name="qSlicerVolumeRenderingModule">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>616</width>
    <height>680</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Volume Render</string>
  </property>
  <layout class="QFormLayout" name="formLayout_2">
   <property name="margin">
    <number>0</number>
   </property>
   <item row="0" column="0">
    <widget class="QLabel" name="VolumeNodeLabel">
     <property name="text">
      <string>Volume:</string>
     </property>
    </widget>
   </item>
   <item row="0" column="1">
    <widget class="qMRMLNodeComboBox" name="VolumeNodeComboBox">
     <property name="nodeTypes">
      <stringlist>
       <string>vtkMRMLScalarVolumeNode</string>
      </stringlist>
     </property>
     <property name="addEnabled">
      <bool>false</bool>
     </property>
     <property name="renameEnabled">
      <bool>true</bool>
     </property>
    </widget>
   </item>
   <item row="2" column="0" colspan="2">
    <widget class="ctkCollapsibleButton" name="InputsCollapsibleButton">
     <property name="enabled">
      <bool>false</bool>
     </property>
     <property name="text">
      <string>Inputs</string>
     </property>
     <property name="collapsed">
      <bool>true</bool>
     </property>
     <layout class="QFormLayout" name="formLayout">
      <item row="1" column="0">
       <widget class="QLabel" name="DisplayNodeLlabel">
        <property name="text">
         <string>Display:</string>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="qMRMLNodeComboBox" name="DisplayNodeComboBox">
        <property name="enabled">
         <bool>false</bool>
        </property>
        <property name="nodeTypes">
         <stringlist>
          <string>vtkMRMLVolumeRenderingDisplayNode</string>
         </stringlist>
        </property>
        <property name="renameEnabled">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item row="2" column="0">
       <widget class="QLabel" name="ROINodeLabel">
        <property name="text">
         <string>ROI:</string>
        </property>
       </widget>
      </item>
      <item row="2" column="1">
       <widget class="qMRMLNodeComboBox" name="ROINodeComboBox">
        <property name="nodeTypes">
         <stringlist>
          <string>vtkMRMLAnnotationROINode</string>
         </stringlist>
        </property>
        <property name="noneEnabled">
         <bool>true</bool>
        </property>
        <property name="renameEnabled">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item row="3" column="0">
       <widget class="QLabel" name="VolumePropertyNodeLabel">
        <property name="text">
         <string>Property:</string>
        </property>
       </widget>
      </item>
      <item row="3" column="1">
       <widget class="qMRMLNodeComboBox" name="VolumePropertyNodeComboBox">
        <property name="nodeTypes">
         <stringlist>
          <string>vtkMRMLVolumePropertyNode</string>
         </stringlist>
        </property>
        <property name="noneEnabled">
         <bool>true</bool>
        </property>
        <property name="renameEnabled">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item row="4" column="0">
       <widget class="QLabel" name="ViewNodeLabel">
        <property name="text">
         <string>View:</string>
        </property>
       </widget>
      </item>
      <item row="0" column="0">
       <widget class="QLabel" name="VisibilityLabel">
        <property name="text">
         <string>Visible:</string>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="QCheckBox" name="VisibilityCheckBox">
        <property name="enabled">
         <bool>false</bool>
        </property>
        <property name="text">
         <string/>
        </property>
       </widget>
      </item>
      <item row="4" column="1">
       <widget class="qMRMLCheckableNodeComboBox" name="ViewCheckableNodeComboBox">
        <property name="nodeTypes">
         <stringlist>
          <string>vtkMRMLViewNode</string>
         </stringlist>
        </property>
        <property name="baseName">
         <string>View</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item row="3" column="0" colspan="2">
    <widget class="ctkCollapsibleButton" name="RenderingCollapsibleButton">
     <property name="text">
      <string>Rendering</string>
     </property>
     <property name="checked">
      <bool>true</bool>
     </property>
     <property name="contentsFrameShape">
      <enum>QFrame::StyledPanel</enum>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout">
      <item>
       <widget class="QTabWidget" name="RenderingTabWidget">
        <property name="currentIndex">
         <number>0</number>
        </property>
        <widget class="QWidget" name="TechniquesTab">
         <attribute name="title">
          <string>Techniques</string>
         </attribute>
         <layout class="QFormLayout" name="formLayout_5">
          <item row="0" column="0">
           <widget class="QLabel" name="RenderingMethodLabel">
            <property name="text">
             <string>Rendering Method:</string>
            </property>
           </widget>
          </item>
          <item row="0" column="1">
           <widget class="QComboBox" name="RenderingMethodComboBox">
            <item>
             <property name="text">
              <string>VTK CPU Ray Casting</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>VTK GPU Ray Casting</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>VTK OpenGL 3D Texture Mapping</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>NCI GPU Ray Casting</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>NCI GPU Ray Casting (Multi-Volume)</string>
             </property>
            </item>
           </widget>
          </item>
          <item row="1" column="0">
           <widget class="QLabel" name="MemorySizeLabel">
            <property name="text">
             <string>GPU Memory Size:</string>
            </property>
           </widget>
          </item>
          <item row="1" column="1">
           <widget class="QComboBox" name="MemorySizeComboBox">
            <property name="currentIndex">
             <number>-1</number>
            </property>
           </widget>
          </item>
          <item row="2" column="0">
           <widget class="QLabel" name="QualityControlLabel">
            <property name="text">
             <string>Quality Control:</string>
            </property>
           </widget>
          </item>
          <item row="2" column="1">
           <widget class="QComboBox" name="QualityControlComboBox">
            <property name="currentIndex">
             <number>0</number>
            </property>
            <item>
             <property name="text">
              <string>Adaptative</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>Maximum Quality</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>Fixed Framerate</string>
             </property>
            </item>
           </widget>
          </item>
          <item row="3" column="0">
           <widget class="QLabel" name="FramerateLabel">
            <property name="text">
             <string>Interactive Speed:</string>
            </property>
           </widget>
          </item>
          <item row="3" column="1">
           <widget class="ctkSliderWidget" name="FramerateSliderWidget">
            <property name="decimals">
             <number>0</number>
            </property>
            <property name="minimum">
             <double>1.000000000000000</double>
            </property>
            <property name="maximum">
             <double>20.000000000000000</double>
            </property>
            <property name="value">
             <double>8.000000000000000</double>
            </property>
            <property name="suffix">
             <string> fps</string>
            </property>
           </widget>
          </item>
          <item row="4" column="0" colspan="2">
           <widget class="ctkCollapsibleGroupBox" name="AdvancedGroupBox">
            <property name="title">
             <string>Advanced Rendering Properties</string>
            </property>
            <property name="checked">
             <bool>false</bool>
            </property>
            <layout class="QFormLayout" name="formLayout_7">
             <property name="fieldGrowthPolicy">
              <enum>QFormLayout::AllNonFixedFieldsGrow</enum>
             </property>
             <property name="margin">
              <number>0</number>
             </property>
             <item row="0" column="0">
              <widget class="QLabel" name="RenderingTechniqueLabel">
               <property name="text">
                <string>Technique (bg):</string>
               </property>
              </widget>
             </item>
             <item row="0" column="1">
              <widget class="QComboBox" name="RenderingTechniqueComboBox">
               <property name="toolTip">
                <string>Select Ray casting technique for the background (default) volume.</string>
               </property>
              </widget>
             </item>
             <item row="1" column="0" colspan="2">
              <widget class="QStackedWidget" name="stackedWidget">
               <property name="currentIndex">
                <number>0</number>
               </property>
               <widget class="QWidget" name="VTKCPUPage">
                <layout class="QFormLayout" name="formLayout_4">
                 <property name="fieldGrowthPolicy">
                  <enum>QFormLayout::AllNonFixedFieldsGrow</enum>
                 </property>
                </layout>
               </widget>
               <widget class="QWidget" name="VKTGPUPage">
                <layout class="QFormLayout" name="formLayout_6"/>
               </widget>
               <widget class="QWidget" name="VTKGPUTexturePage"/>
               <widget class="QWidget" name="NCIGPUPage">
                <layout class="QFormLayout" name="formLayout_8">
                 <property name="margin">
                  <number>0</number>
                 </property>
                 <item row="0" column="0">
                  <widget class="QLabel" name="DistColorBlendingLabel">
                   <property name="text">
                    <string>Dist. Color Blending:</string>
                   </property>
                  </widget>
                 </item>
                 <item row="0" column="1">
                  <widget class="ctkSliderWidget" name="DistColorBlendingSliderWidget">
                   <property name="toolTip">
                    <string>Distance Color Blending: Voxels with longer distance to eye/camera would be more darker to reveal depth information in volume. Higher value indicates stronger darking effect. Setting the value to 0 will turn off the effect.</string>
                   </property>
                   <property name="singleStep">
                    <double>0.100000000000000</double>
                   </property>
                   <property name="pageStep">
                    <double>0.200000000000000</double>
                   </property>
                   <property name="maximum">
                    <double>1.000000000000000</double>
                   </property>
                  </widget>
                 </item>
                 <item row="1" column="1">
                  <widget class="ctkSliderWidget" name="ICPEScaleSliderWidget">
                   <property name="toolTip">
                    <string>Parameter scale (kt) for Illustrative Context Preserving Exploration, which simulates depth penetrating into volume.</string>
                   </property>
                   <property name="singleStep">
                    <double>0.100000000000000</double>
                   </property>
                   <property name="pageStep">
                    <double>4.000000000000000</double>
                   </property>
                   <property name="maximum">
                    <double>20.000000000000000</double>
                   </property>
                  </widget>
                 </item>
                 <item row="2" column="1">
                  <widget class="ctkSliderWidget" name="ICPESmoothnessSliderWidget">
                   <property name="toolTip">
                    <string>Parameter smoothness (ks) for Illustrative Context Preserving Exploration.</string>
                   </property>
                   <property name="singleStep">
                    <double>0.100000000000000</double>
                   </property>
                   <property name="pageStep">
                    <double>0.200000000000000</double>
                   </property>
                   <property name="maximum">
                    <double>1.000000000000000</double>
                   </property>
                   <property name="value">
                    <double>0.500000000000000</double>
                   </property>
                  </widget>
                 </item>
                 <item row="3" column="1">
                  <widget class="ctkSliderWidget" name="DepthPeelingSliderWidget">
                   <property name="toolTip">
                    <string>Set threshold for Volumetric Depth Peeling. Volume rendering starts after we have met scalar values higher than the threshold. Use with transfer functions together.</string>
                   </property>
                  </widget>
                 </item>
                 <item row="1" column="0">
                  <widget class="QLabel" name="ICPEScaleLabel">
                   <property name="text">
                    <string>ICPE Scale:</string>
                   </property>
                  </widget>
                 </item>
                 <item row="2" column="0">
                  <widget class="QLabel" name="ICPESmoothnessLabel">
                   <property name="text">
                    <string>ICPE: Smoothness:</string>
                   </property>
                  </widget>
                 </item>
                 <item row="3" column="0">
                  <widget class="QLabel" name="DepthPeelingLabel">
                   <property name="text">
                    <string>Vol. Depth Peeling:</string>
                   </property>
                  </widget>
                 </item>
                </layout>
               </widget>
               <widget class="QWidget" name="NCIGPUMultiPage">
                <layout class="QFormLayout" name="formLayout_9">
                 <property name="margin">
                  <number>0</number>
                 </property>
                 <item row="0" column="0">
                  <widget class="QLabel" name="RenderingTechniqueFgLabel">
                   <property name="text">
                    <string>Technique (fg):</string>
                   </property>
                  </widget>
                 </item>
                 <item row="0" column="1">
                  <widget class="QComboBox" name="RenderingTechniqueFgComboBox">
                   <property name="toolTip">
                    <string>Select GPU ray casting technique for fg volume.</string>
                   </property>
                  </widget>
                 </item>
                 <item row="1" column="0">
                  <widget class="QLabel" name="FusionLabel">
                   <property name="text">
                    <string>Fusion:</string>
                   </property>
                  </widget>
                 </item>
                 <item row="1" column="1">
                  <widget class="QComboBox" name="FusionComboBox">
                   <property name="toolTip">
                    <string>Select color fusion method in multi-volume rendering</string>
                   </property>
                   <item>
                    <property name="text">
                     <string>Alpha Blending (OR)</string>
                    </property>
                   </item>
                   <item>
                    <property name="text">
                     <string>Alpha Blending (AND)</string>
                    </property>
                   </item>
                   <item>
                    <property name="text">
                     <string>Alpha Blending (NOT)</string>
                    </property>
                   </item>
                  </widget>
                 </item>
                 <item row="2" column="0">
                  <widget class="QLabel" name="BgFgRatioLabel">
                   <property name="text">
                    <string>Bg|Fg ratio:</string>
                   </property>
                  </widget>
                 </item>
                 <item row="2" column="1">
                  <widget class="ctkSliderWidget" name="BgFgRatioSliderWidget">
                   <property name="singleStep">
                    <double>0.100000000000000</double>
                   </property>
                   <property name="pageStep">
                    <double>0.200000000000000</double>
                   </property>
                   <property name="maximum">
                    <double>1.000000000000000</double>
                   </property>
                   <property name="value">
                    <double>0.500000000000000</double>
                   </property>
                  </widget>
                 </item>
                </layout>
               </widget>
              </widget>
             </item>
            </layout>
           </widget>
          </item>
         </layout>
        </widget>
        <widget class="QWidget" name="VolumePropertyTab">
         <attribute name="title">
          <string>Volume Properties</string>
         </attribute>
         <layout class="QVBoxLayout" name="volumePropertyLayout">
          <property name="margin">
           <number>0</number>
          </property>
          <item>
           <widget class="qMRMLVolumePropertyNodeWidget" name="VolumePropertyNodeWidget"/>
          </item>
         </layout>
        </widget>
        <widget class="QWidget" name="ROITab">
         <attribute name="title">
          <string>ROI</string>
         </attribute>
         <layout class="QFormLayout" name="formLayout_3">
          <property name="fieldGrowthPolicy">
           <enum>QFormLayout::AllNonFixedFieldsGrow</enum>
          </property>
          <item row="0" column="0">
           <widget class="QLabel" name="ROICropLabel">
            <property name="text">
             <string>Crop:</string>
            </property>
           </widget>
          </item>
          <item row="0" column="1">
           <widget class="QCheckBox" name="ROICropCheckBox">
            <property name="text">
             <string/>
            </property>
           </widget>
          </item>
          <item row="1" column="0" colspan="2">
           <widget class="qMRMLAnnotationROIWidget" name="ROIWidget"/>
          </item>
          <item row="2" column="0" colspan="2">
           <widget class="QPushButton" name="ROIFitPushButton">
            <property name="text">
             <string>Fit ROI to Volume</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
        <widget class="QWidget" name="MiscTab">
         <attribute name="title">
          <string>Misc</string>
         </attribute>
        </widget>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>qMRMLCheckableNodeComboBox</class>
   <extends>qMRMLNodeComboBox</extends>
   <header>qMRMLCheckableNodeComboBox.h</header>
  </customwidget>
  <customwidget>
   <class>qMRMLNodeComboBox</class>
   <extends>QWidget</extends>
   <header>qMRMLNodeComboBox.h</header>
  </customwidget>
  <customwidget>
   <class>qMRMLVolumePropertyNodeWidget</class>
   <extends>QWidget</extends>
   <header>qMRMLVolumePropertyNodeWidget.h</header>
  </customwidget>
  <customwidget>
   <class>qMRMLAnnotationROIWidget</class>
   <extends>QWidget</extends>
   <header>qMRMLAnnotationROIWidget.h</header>
  </customwidget>
  <customwidget>
   <class>qSlicerWidget</class>
   <extends>QWidget</extends>
   <header>qSlicerWidget.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>ctkCollapsibleButton</class>
   <extends>QWidget</extends>
   <header>ctkCollapsibleButton.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>ctkCollapsibleGroupBox</class>
   <extends>QGroupBox</extends>
   <header>ctkCollapsibleGroupBox.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>ctkSliderWidget</class>
   <extends>QWidget</extends>
   <header>ctkSliderWidget.h</header>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="../../../Annotation/Resources/qSlicerAnnotationModule.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>VolumePropertyNodeComboBox</sender>
   <signal>currentNodeChanged(vtkMRMLNode*)</signal>
   <receiver>VolumePropertyNodeWidget</receiver>
   <slot>setMRMLVolumePropertyNode(vtkMRMLNode*)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>396</x>
     <y>81</y>
    </hint>
    <hint type="destinationlabel">
     <x>180</x>
     <y>456</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>ROINodeComboBox</sender>
   <signal>currentNodeChanged(vtkMRMLNode*)</signal>
   <receiver>ROIWidget</receiver>
   <slot>setMRMLAnnotationROINode(vtkMRMLNode*)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>606</x>
     <y>79</y>
    </hint>
    <hint type="destinationlabel">
     <x>257</x>
     <y>314</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>VolumeNodeComboBox</sender>
   <signal>currentNodeChanged(bool)</signal>
   <receiver>InputsCollapsibleButton</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>154</x>
     <y>18</y>
    </hint>
    <hint type="destinationlabel">
     <x>307</x>
     <y>64</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>DisplayNodeComboBox</sender>
   <signal>currentNodeChanged(bool)</signal>
   <receiver>ROINodeComboBox</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>315</x>
     <y>77</y>
    </hint>
    <hint type="destinationlabel">
     <x>320</x>
     <y>79</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>DisplayNodeComboBox</sender>
   <signal>currentNodeChanged(bool)</signal>
   <receiver>VolumePropertyNodeComboBox</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>296</x>
     <y>77</y>
    </hint>
    <hint type="destinationlabel">
     <x>298</x>
     <y>81</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>DisplayNodeComboBox</sender>
   <signal>currentNodeChanged(bool)</signal>
   <receiver>VisibilityCheckBox</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>546</x>
     <y>77</y>
    </hint>
    <hint type="destinationlabel">
     <x>520</x>
     <y>75</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>DisplayNodeComboBox</sender>
   <signal>currentNodeChanged(bool)</signal>
   <receiver>ViewCheckableNodeComboBox</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>371</x>
     <y>77</y>
    </hint>
    <hint type="destinationlabel">
     <x>372</x>
     <y>83</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>VolumeNodeComboBox</sender>
   <signal>currentNodeChanged(bool)</signal>
   <receiver>InputsCollapsibleButton</receiver>
   <slot>setChecked(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>145</x>
     <y>18</y>
    </hint>
    <hint type="destinationlabel">
     <x>149</x>
     <y>58</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>qSlicerVolumeRenderingModule</sender>
   <signal>mrmlSceneChanged(vtkMRMLScene*)</signal>
   <receiver>ViewCheckableNodeComboBox</receiver>
   <slot>setMRMLScene(vtkMRMLScene*)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>37</x>
     <y>260</y>
    </hint>
    <hint type="destinationlabel">
     <x>289</x>
     <y>83</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>qSlicerVolumeRenderingModule</sender>
   <signal>mrmlSceneChanged(vtkMRMLScene*)</signal>
   <receiver>ROINodeComboBox</receiver>
   <slot>setMRMLScene(vtkMRMLScene*)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>438</x>
     <y>228</y>
    </hint>
    <hint type="destinationlabel">
     <x>606</x>
     <y>79</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>qSlicerVolumeRenderingModule</sender>
   <signal>mrmlSceneChanged(vtkMRMLScene*)</signal>
   <receiver>VolumePropertyNodeComboBox</receiver>
   <slot>setMRMLScene(vtkMRMLScene*)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>548</x>
     <y>227</y>
    </hint>
    <hint type="destinationlabel">
     <x>606</x>
     <y>81</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>qSlicerVolumeRenderingModule</sender>
   <signal>mrmlSceneChanged(vtkMRMLScene*)</signal>
   <receiver>DisplayNodeComboBox</receiver>
   <slot>setMRMLScene(vtkMRMLScene*)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>389</x>
     <y>228</y>
    </hint>
    <hint type="destinationlabel">
     <x>602</x>
     <y>77</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>qSlicerVolumeRenderingModule</sender>
   <signal>mrmlSceneChanged(vtkMRMLScene*)</signal>
   <receiver>VolumeNodeComboBox</receiver>
   <slot>setMRMLScene(vtkMRMLScene*)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>501</x>
     <y>228</y>
    </hint>
    <hint type="destinationlabel">
     <x>445</x>
     <y>19</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>RenderingMethodComboBox</sender>
   <signal>currentIndexChanged(int)</signal>
   <receiver>stackedWidget</receiver>
   <slot>setCurrentIndex(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>300</x>
     <y>169</y>
    </hint>
    <hint type="destinationlabel">
     <x>169</x>
     <y>329</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
